File: P3/QytetetJava/src/modeloQytetet/TituloPropiedad.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloQytetet;

import java.util.ArrayList;

/**
 *
 * @author marquirj
 */
public class TituloPropiedad {
    private String nombre;
    private boolean hipotecado=false;
    private int precioCompra;
    private int alquilerBase;
    private float factorRevalorizacion;
    private int hipotecaBase;
    private int precioEdificar;
    private int numCasas=0;
    private int numHoteles=0;
    private Jugador propietario;
    TituloPropiedad(String nombre, int preciocompra, int alquilerbase, float factorrevalorizacion, int hipotecabase, int precioedificar){
        this.nombre=nombre;
        this.hipotecado=false;
        this.precioCompra=preciocompra;
        this.alquilerBase=alquilerbase;
        this.factorRevalorizacion=factorrevalorizacion;
        this.hipotecaBase=hipotecabase;
        this.precioEdificar=precioedificar;
        this.numCasas=0;
        this.numHoteles=0;
    }

    public String getNombre() {
        return nombre;
    }

    public boolean isHipotecado() {
        return hipotecado;
    }

    public int getPrecioCompra() {
        return precioCompra;
    }

    public int getAlquilerBase() {
        return alquilerBase;
    }

    public float getFactorRevalorizacion() {
        return factorRevalorizacion;
    }

    public int getHipotecaBase() {
        return hipotecaBase;
    }

    public int getPrecioEdificar() {
        return precioEdificar;
    }

    public int getNumCasas() {
        return numCasas;
    }

    public int getNumHoteles() {
        return numHoteles;
    }

    public void setHipotecado(boolean hipotecado) {
        this.hipotecado = hipotecado;
    }
    int calcularCosteCancelar(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    int calcularCostHipotecar(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    int calcularImporteAlquiler(){
        throw new UnsupportedOperationException("Sin implementar");
        
    }
    int calcularPrecioVenta(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    void cancelarHipoteca(){}
    void cobrarAlquiler(int coste){}
    void edificarCasa(){}
    void edificarHotel(){}

    public Jugador getPropietario() {
        return propietario;
    }
    int pagarAlquiler(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    boolean propietarioEncarcelado(){
        throw new UnsupportedOperationException("Sin implementar");
    }
    void setHipotecada(boolean hipotecada){}

    public void setPropietario(Jugador propietario) {
        this.propietario = propietario;
    }
    boolean tengoPropietario(){
        throw new UnsupportedOperationException("Sin implementar");
    }
        
    
    
    
    
    
    
    
    
    
    
    

    @Override
    public String toString() {
        return "TituloPropiedad{" + "nombre=" + nombre + ", hipotecado=" + hipotecado + ", precioCompra=" + precioCompra + ", alquilerBase=" + alquilerBase + ", factorRevalorizacion=" + factorRevalorizacion + ", hipotecaBase=" + hipotecaBase + ", precioEdificar=" + precioEdificar + ", numCasas=" + numCasas + ", numHoteles=" + numHoteles + '}';
    }
    
}

File: P3/QytetetRuby/lib/titulo_propiedad.rb
#ENCODING: UTF-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module QytetetRuby
class TituloPropiedad
  attr_accessor :hipotecada, :propietario
  attr_reader :nombre, :precioCompra, :alquilerBase, :factorRevalorizacion, :hipotecaBase, :precioEdificar, :numCasas, :numHoteles
  def initialize(nombre_,precioCompra_,alquilerBase_,factorRevalorizacion_,hipotecaBase_,precioEdificar_)
   @nombre=nombre_
   @hipotecada=false
   @precioCompra=precioCompra_
   @alquilerBase=alquilerBase_
   @factorRevalorizacion=factorRevalorizacion_
   @hipotecaBase=hipotecaBase_
   @precioEdificar=precioEdificar_
   @numCasas=5
   @numHoteles=6
   @propietario=nil
  end
  def calcularCosteCancelar
    raise NotImplementedError
  end
  def calcularCosteHipotecar
    raise NotImplementedError
  end
  def calcularCosteAlquiler
    raise NotImplementedError
  end
  def calcularPrecioVenta
    raise NotImplementedError
  end
  def cancelarHipoteca
    raise NotImplementedError
  end
  def cobrarAlquiler(alquiler_)
    raise NotImplementedError
  end
  def edificaCasa
    raise NotImplementedError
  end
  def edificaHotel
    raise NotImplementedError
  end
  def hipotecar
    raise NotImplementedError
  end
  def pagarAlquiler
    raise NotImplementedError
  end
  def propietarioEncarcelado
    return @propietario.encarcelado
  end
  def tengoPropietario
    return @propietario!=nil
  end

  def to_s
    "Nombre : #{@nombre}\n
     Hipotecada?: #{@hipotecada} \n
     Precio compra: #{@precioCompra}\n
     Alquiler Base: #{@alquilerBase}\n
     Factor Revalorizacion: #{@factorRevalorizacion} \n
     Hipoteca Base: #{@hipotecaBase}\n
     Precio Edificar : #{@precioEdificar}\n
     Num Casas: #{@numCasas} \n
     Num Hoteles: #{@numHoteles}\n
     Propietario: #{@propietario}"
  end
end
end